### PR TITLE
Radial menu for metal recipes + small cleanup of mats code

### PR DESCRIPTION
**Pull request by breadhunt**
## About The Pull Request

Continuation of https://github.com/tgstation/TerraGov-Marine-Corps/pull/10315 , merge conflicts be damned

- Adds a quick-access radial menu for metal. The old recipe menu can be accessed through the + button:
image
(Feel free to suggest changes to it)

- Adds a toggle to the radial menu. Works like medical radial wheel, if you disable it, it'll just work as before.

- Removes plasteel box recipe. If any other recipes are added for plasteel feel free to re-add. Plasteel cades can now be made in the same way as sandbags.
- Cleans up sandbag code while we're at it

## Why It's Good For The Game

Old recipe menu was clunky (had to click on it to build cade, click off to move, click back on to build cade again) and obtrusive for beginners.
This makes it easier for new engis to use the menu, and reduces clicks needed.

I don't think it's too hard to access old recipe menu from this (it's one extra click) but if it really becomes an issue I can make it a preference like radial medical wheel. I might also make TGUI recipe menu after this but TGUI is hard and I am lazy, so no promises. Please give me your opinions engi mains.

## Changelog
:cl:
del: Removed plasteel crate recipe
qol: Plasteel cades no longer require a menu to make, just press Z
qol: Metal recipes have a new radial menu, old recipe menu selectable from it
add: You can also disable the radial menu completely for the old menu in the preferences tab
refactor: Cleaned up mat and sandbag code a small bit
/:cl:

### CODE REVIEW

**Comment by Lauwma** [comment]
Speedmerge

**Comment by Losenis** [comment]
Can you add the old menu accessible through right clicking the stack? Otherwise tying it to two actions on top of ping is gonna be cancer for those who want to use the old menu regardless.

**Comment by breadhunt** [comment]
> Can you add the old menu accessible through right clicking the stack? Otherwise tying it to two actions on top of ping is gonna be cancer for those who want to use the old menu regardless.

isn't right click already used to separate stacks? I'll see if I can get it on ctrl right click

**Comment by Derrick2Electricboogaloo** [comment]
So long as I can still access the Z menu for metal this seems really nice.

**Comment by breadhunt** [comment]
@Derrick2Electricboogaloo @Losenis toggle added in preferences. Like medical wheel, if you disable it, it'll just work as before

**Review by TiviPlus — CHANGES_REQUESTED** [review]
why did you add the /obj/item/stack/attack_self(mob/user) code if you dont call parent anyway

**Review by Kuro020 — APPROVED** [review]

File: code/modules/client/preferences_toggles.dm [hunk 1]
```diff
@@ -97,6 +97,14 @@
...
 	to_chat(src, span_notice("You will [(prefs.toggles_gameplay & RADIAL_MEDICAL) ? "now" : "no longer"] use the radial menu for medical purposes."))
 
+/client/verb/toggle_radial_stacks()
+	set category = "Preferences"
+	set name = "Toggle Radial Stacks Wheel"
+
+	prefs.toggles_gameplay ^= RADIAL_STACKS
+	prefs.save_preferences()
+
+	to_chat(src, span_notice("You will [(prefs.toggles_gameplay & RADIAL_STACKS) ? "now" : "no longer"] use the radial menu when interacting with material stacks."))
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/client/preferences_toggles.dm:107]: if you do not in this pr are then

**Comment by TiviPlus** [comment]
Also adding here but I said in discord that you need to call parent first in attack self due to signal

**Comment by Durbles** [comment]
Very good change yes. I can't count how many times I've made razorwire instead of barbed wire because of how clunky the current menu is.

**Pull request merged**